### PR TITLE
Skip errors during harvest and report to Sentry

### PR DESCRIPTION
**Pull request by ghukill**
#### What does this PR do?

Allows graceful skip of errors during `method=get` / `GetRecords` style harvest.

If errors are encountered, and the count is below a new config value `MAX_ALLOWED_ERRORS`, a message is sent to Sentry that looks roughly like:

Screenshot 2023-11-15 at 4 37 15 PM

However, if more errors are encountered than defined by `MAX_ALLOWED_ERRORS`, the harvested is stopped and a message like the following will be sent to Sentry:

Screenshot 2023-11-15 at 4 42 21 PM

In both scenarios, an additional data point `failed_records` will be present in the Sentry message that shows which records failed, e.g.:

Screenshot 2023-11-15 at 4 37 40 PM

#### Helpful background context

As noted in both TIMX-257 and TIMX-258 (linked below), until now an external "skip list" has been used to skip records that were consistently throwing errors.  This was functional, but has some overhead for maintaining that list; more detail can be found in the tickets.

#### How can a reviewer manually see the effects of these changes?

The most straightforward way is to perform a local harvest that targets some known (at time of writing) problematic records:

```python
from harvester.oai import OAIClient

oai_client = OAIClient(
    "https://dspace.mit.edu/oai/request",
    metadata_format="oai_dc",
    retry_status_codes=tuple() # sets to empty tuple to avoid numerous retries
)
identifiers = [
    "oai:dspace.mit.edu:1721.1/152958",
    "oai:dspace.mit.edu:1721.1/152786", # BAD
    "oai:dspace.mit.edu:1721.1/147573", # BAD
    "oai:dspace.mit.edu:1721.1/152939",
]
records = list(oai_client.get_records(identifiers))
```

This should complete, but `records` should only be length 2 for the two that completed successfully.  

To confirm that messages are sent to Sentry, you can do the following:
  * set `.env`

```
WORKSPACE=dev
SENTRY_DSN=<ADD HERE>
```

And then prefix the following to python code above:
```python
from harvester.config import configure_sentry
configure_sentry()

# rest of code...
```

After re-running, there should be a new message [here in Sentry](https://mit-libraries.sentry.io/issues/?project=6390409&query=is%3Aunresolved&referrer=issue-list&statsPeriod=24h).

#### Includes new or updated dependencies?

YES

NOTE: both `vcrpy` and `urllib3` was pinned until work is completed for this issue: https://github.com/MITLibraries/oai-pmh-harvester/issues/550

#### What are the relevant tickets?

  * https://mitlibraries.atlassian.net/browse/TIMX-257
  * https://mitlibraries.atlassian.net/browse/TIMX-258

#### Developer checklist

- [X] All new ENV is documented in README
- [X] Stakeholder approval has been confirmed (or is not needed)

#### Code reviewer checklist

- [x] The commit message is clear and follows our guidelines (not just this pull request message)
- [x] There are appropriate tests covering any new functionality
- [x] The documentation has been updated or is unnecessary
- [x] The changes have been verified
- [x] New dependencies are appropriate or there were no changes

### CODE REVIEW

File: harvester/utils.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+def send_sentry_message(
+    message: str,
+    scopes: dict | None = None,
+    level: str = "warning",
+):
+    """Send message directly to Sentry.
+
+    This allows both reporting information without raising an Exception, and optionally
+    including additional information in the sent Sentry message via "scopes".
```
Comment (ehanson8) [inline harvester/utils.py:14]: Description of the args would be helpful, `scopes` isn't very clear on its own

**Review by ehanson8 — APPROVED** [review]
Looks good and works as expected, one suggestion!

File: harvester/utils.py (hunk 1 above)
Comment (ghukill) [reply harvester/utils.py:14]: Good call!  Done.

File: harvester/utils.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    :param message: primary message string for Sentry message
```
Comment (ehanson8) [inline harvester/utils.py:16]: Nitpicky, but we've usually done: 

```
Args:
    message: Primary message string for Sentry message.
    scopes: Dictionary of key/value pairs which become additional information in the Sentry message.
    level: String of [info,debug,warning,error] that will set the severity of the  message in Sentry.
"""
```

**Pull request merged**